### PR TITLE
differentiate serial numbers from lot management

### PR DESCRIPTION
**Pull request by altairis-tof**
add a value to status_batch for a better serial numbers management in batch products.

This is the first commit of many that will lead to better trace equipments with unique product serial numbers.

It was analysed in conjonction with @FHenry and @lmag :

Now there are 3 status on a product : 
0 : No (lot/serial not used)
1 : Lot (required)
2 : Serial number (required)

### CODE REVIEW

**Comment by altairis-tof** [comment]
@eldy Hello Laurent, stickler has first rejected my PR because of missing space after else statement; i fixed it but stickler still fails and i do not understand why.

**Comment by frederic34** [comment]
@altatof it happen sometimes

**Comment by altairis-tof** [comment]
@frederic34 do i have to do anything ?

**Comment by frederic34** [comment]
nothing for the moment, just wait for @eldy advice :crossed_fingers:

**Comment by eldy** [comment]
@altatof 
Can you confirm what you mean with value 2 ?
The value Log/Serial must have 3 values:
0=No lot/serial number
1=Lot/serial number required
2=Serial number (must be unique for each copy of product)

Is it what you mean with value 2 ? In your implementation, it seems there is no difference between 1 and 2. Goal is to have a value 2 that differs with value 1 because when product has value 2, each instance/copy of the product must have a different number (it means size of each lot is 1). Is it what you expect to do ?
If you add a check on unicity when we insert a new record on the number, then we can introduce the value 2, with no need to introduce a hidden option for that. Label must be clear that difference is that each copy has a unique number when choosing type 2.

**Comment by altairis-tof** [comment]
Ok @eldy it's exactly what we want to do; next PR will be check for unicity of the serial number and also ability to add masks to define serial numbers automatically

**Comment by altairis-tof** [comment]
i will remove hidden constant in next commit when the check of unicity will be coded

**Comment by fappels** [comment]
I thought lot number used to be unique for a product in the past (unique constraint on product - batch - eatby - sellby), sure there is no regression to fix?

**Comment by eldy** [comment]
@fappels Right there was a constraint in the past. But here, we will not add any constraint. We will just add a third status for lot and a check will be done when the lot/serial = 2 for the product that : 
After the stock movement, the quantity for this lot and product, in all warehouses, remains to 0 or 1
This is an applicative stock check only so we won't be able to have into stock a quantity higher than 1 for this lot/serial.

**Comment by altairis-tof** [comment]
:+1: :)

File: htdocs/product/list.php [hunk 1]
```diff
@@ -833,7 +833,21 @@
...
+		if ( empty($conf->global ->MAIN_ADVANCE_NUMLOT) )
+		{
+			print $form->selectyesno('search_tobatch', $search_tobatch, 1, false, 1);
+		} else
+		{
+			$statutarray = array('-1' => '', '0' => $langs->trans("ProductStatusNotOnBatchShort"), '1' => $langs->trans("ProductStatusOnBatchShort"), '2' => $langs->trans("ProductStatusOnSerialShort"));
+			print $form->selectarray('search_tobatch', $statutarray, $search_tobatch);
+		}
+
+		'</td>';
```
Comment (frederic34) [inline htdocs/product/list.php:849]: print ?

Comment (frederic34) [reply htdocs/product/list.php:849]: I make a commit for that

**Pull request merged**